### PR TITLE
Week 3 branch memoize function

### PR DESCRIPTION
**Pull request by Maheima**

### CODE REVIEW

File: Week-3/assignment3_4.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      return this.items.pop();
+    },
+  };
+}
+
+const stack = createStack();
+stack.push(10);
+stack.push(5);
+stack.pop(); 
+console.log(stack.items);
```
Comment (jesusverma) [inline Week-3/assignment3_4.js:20]: you can still access items outside of createStack. Please modify the code. This code seems as it is mentioned in assignment.

File: Week-3/assignment3_3.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        count++;
+    }
+    let message=`Count is${count}`;
+    
+    function log() {
+        console.log(message);
+    }
+    return[increment,log];
+}
+const [increment,log] = createIncrement();
```
Comment (jesusverma) [inline Week-3/assignment3_3.js:13]: How this line number 13 is working please explain ?

Comment (Maheima) [reply Week-3/assignment3_3.js:13]: We are destructuring the array here

File: Week-3/assignment3_4.js (hunk 1 above)
Comment (Maheima) [reply Week-3/assignment3_4.js:20]: Let me check

**Pull request merged**